### PR TITLE
fix: test output under debugger

### PR DESCRIPTION
**Pull request by yury-s**
Fixes https://github.com/microsoft/playwright/issues/14372

### CODE REVIEW

File: src/oopReporter.ts [hunk 1]
```diff
@@ -71,6 +71,19 @@ class OopReporter implements Reporter {
       };
       t.onclose = () => process.exit(0);
     });
+
+    if (process.env.PW_OUT_OF_PROCESS_REPORTER_UNDER_DEBUGGER) {
```
Comment (pavelfeldman) [inline src/oopReporter.ts:75]: Make this unconditional?

File: src/oopReporter.ts [hunk 2]
```diff
@@ -148,6 +161,19 @@ class OopReporter implements Reporter {
     this._emit('onStepEnd', params);
   }
 
+  onStdOut(chunk: string | Buffer) {
```
Comment (pavelfeldman) [inline src/oopReporter.ts:164]: Don't add these, rely on underlying reporters to produce output handled above.

File: src/playwrightTest.ts [hunk 3]
```diff
@@ -215,15 +215,16 @@ export class PlaywrightTest {
         PW_TEST_REPORTER: require.resolve('./oopReporter'),
         PW_TEST_REPORTER_WS_ENDPOINT: wsEndpoint,
         PW_TEST_HTML_REPORT_OPEN: 'never',
+        PW_OUT_OF_PROCESS_REPORTER_UNDER_DEBUGGER: 1,
```
Comment (pavelfeldman) [inline src/playwrightTest.ts:218]: remove this

**Review by pavelfeldman — COMMENTED** [review]
In addition to these, remove lines 169-172

File: src/oopReporter.ts [hunk 4]
```diff
@@ -71,6 +71,19 @@ class OopReporter implements Reporter {
       };
       t.onclose = () => process.exit(0);
     });
+
+    if (process.env.PW_OUT_OF_PROCESS_REPORTER_UNDER_DEBUGGER) {
+      const originalOut = process.stdout.write.bind(process.stdout);
+      process.stdout.write = (chunk: string | Buffer, ...args: any) => {
+        this._emit('onStdOut', { message: chunk.toString() });
+        return originalOut(chunk, ...args);
```
Comment (pavelfeldman) [inline src/oopReporter.ts:79]: Don't call original

File: src/oopReporter.ts [hunk 5]
```diff
@@ -71,6 +71,19 @@ class OopReporter implements Reporter {
...
+      };
+      const originalErr = process.stderr.write.bind(process.stderr);
+      process.stderr.write = (chunk: string | Buffer, ...args: any) => {
+        this._emit('onStdErr', { message: chunk.toString() });
+        return originalErr(chunk, ...args);
```
Comment (pavelfeldman) [inline src/oopReporter.ts:84]: ditto

**Comment by yury-s** [comment]
After a discussion closing this in favor of https://github.com/microsoft/playwright/pull/14455. The output will only appear in the Debug Console under debugger (but not in the test run terminal) which should be good enough for debugging.

**Pull request closed without merging**